### PR TITLE
MLPAB-1544: Add paper-form-received event

### PR DESCRIPTION
**Pull request by acsauk**
Fixes [MLPAB-1544](https://opgtransform.atlassian.net/browse/MLPAB-1544)

[MLPAB-1544]: https://opgtransform.atlassian.net/browse/MLPAB-1544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hawx — APPROVED** [review]
looks fine from my point of view

File: domains/POAS/events/lpa-updated/schema.json [hunk 1]
```diff
@@ -9,10 +9,69 @@
...
+    },
     "changeType": {
       "type": "string",
-      "description": "The type of change to the LPA"
+      "description": "The type of change to the LPA",
+      "enum": ["CREATED", "CERTIFICATE_PROVIDER_SIGN", "ATTORNEY_SIGN", "TRUST_CORPORATION_SIGN"]
     }
   },
-  "required": ["uid", "changeType"]
+  "allOf": [
```
Comment (acsauk) [inline domains/POAS/events/lpa-updated/schema.json:23]: This is gross but doing an `if changeType == CREATED then required = ["uid", "changeType"] else required = ["uid", "actorUID", "changeType"]` was skipping validation on the `changeType` itself. This way catches all validations together.

File: test/validate-examples.mjs [hunk 2]
```diff
@@ -11,30 +11,34 @@ addFormats(ajv);
...
-const examples = await glob(
-  path.join(path.dirname(__filename), "/../domains/**/examples/*.json")
-);
-
-examples.forEach((examplePath) => {
-  const schemaPath = path.join(
-    path.dirname(path.dirname(examplePath)),
-    "schema.json"
-  );
+const schemas = await glob(
```
Comment (acsauk) [inline test/validate-examples.mjs:14]: Looping over examples meant we tried to validate schemas twice if more than one example per schema existed. This way we can add multiple examples which is (sometimes) useful.

File: domains/POAS/events/lpa-updated/schema.json [hunk 3]
```diff
@@ -9,10 +9,69 @@
...
+    "actorUID": {
+      "type": "string",
+      "description": "The UID of the actor the change relates to",
+      "pattern": "urn:opg:poas:makeregister:users:([a-z0-9]{8}-)([a-z0-9]{4}-){3}([a-z0-9]{12})"
+    },
     "changeType": {
       "type": "string",
-      "description": "The type of change to the LPA"
+      "description": "The type of change to the LPA",
+      "enum": ["CREATED", "CERTIFICATE_PROVIDER_SIGN", "ATTORNEY_SIGN", "TRUST_CORPORATION_SIGN"]
```
Comment (hawx) [inline domains/POAS/events/lpa-updated/schema.json:20]: if we've not started using these yet, could we have them as *_SIGNED to keep the tenses the same 👨‍🏫

File: domains/POAS/events/lpa-updated/schema.json [hunk 4]
```diff
@@ -9,10 +9,69 @@
...
+          "changeType": { "const": "ATTORNEY_SIGN" }
+        }
+      },
+      "then": {
+        "properties": {
+          "changeType": { "const": "ATTORNEY_SIGN" }
+        },
+        "required": ["uid", "actorUID", "changeType"]
+      }
+    },
```
Comment (hawx) [inline domains/POAS/events/lpa-updated/schema.json:62]: this should be trust corporation

File: domains/POAS/events/lpa-updated/schema.json (hunk 3 above)
Comment (gregtyler) [reply domains/POAS/events/lpa-updated/schema.json:20]: They're [passthrough from the change events](https://github.com/ministryofjustice/opg-data-lpa-store/blob/f13f291aef41dabdd2f2a6d38aa1e8be5a9a2fce/lambda/update/validate.go), so if we changed them they wouldn't make sense the other way around 😄 

We could switch to `CREATE` so they're all in present tense?

**Review by hawx — APPROVED** [review]

**Pull request merged**